### PR TITLE
fix: Axios issue in Cloudflare workers

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Hopefully fixes https://github.com/PostHog/posthog-js-lite/issues/127
And also https://github.com/PostHog/posthog-js-lite/issues/129

## Changes

The current version of Axios fixes a security issue but has other issues regarding environments like Cloudflare.
Tried to look for a general fix here but the world of polyfilled Fetch is a mess to say the least... In the end, the workaround that made most sense was to simply use fetch if it is defined (which it is in custom environments like Cloudflare workers.

Also added an example project for testing this - managed to recreate the issue and this PR fixes it.

### Bump level

- [ ] Major
- [x] Minor
- [ ] Patch

### Libraries affected

- [ ] All of them
- [ ] posthog-web
- [x] posthog-node
- [ ] posthog-react-native

### Changelog notes
- Fixes issues with Axios imports for non-node environments like Cloudflare workers
- Uses the globally defined `fetch` if available, otherwise imports and uses axios as a polyfill

### CODE REVIEW

**Review by pauldambra — APPROVED** [review]
didn't run it but looks good to me...

a few nit-picks that are very not-blocking

File: posthog-node/test/feature-flags.spec.ts [hunk 1]
```diff
@@ -3,9 +3,9 @@
 import { PostHog as PostHog, PostHogOptions } from '../src/posthog-node'
 import { matchProperty, InconclusiveMatchError } from '../src/feature-flags'
 jest.mock('../src/fetch')
-import { fetch } from '../src/fetch'
+import fetch from '../src/fetch'
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply posthog-node/test/feature-flags.spec.ts:6]: I couldn't actually 😅 As I have to check the global scope with out `global` for cloudflare, if we assign `fetch` anywhere, even as a named export, it gets hoisted up and overrides the global definition. Hence why I changed it to a default export 🤷

[the remark this answers is not in the document]
Comment (benjackwhite) [reply posthog-node/test/feature-flags.spec.ts:6]: I could add a comment to that effect...

[the remark this answers is not in the document]
Comment (benjackwhite) [reply posthog-node/test/feature-flags.spec.ts:6]: Added a comment

**Review by marandaneto — APPROVED** [review]

File: posthog-node/test/feature-flags.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (pauldambra) [reply posthog-node/test/feature-flags.spec.ts:6]: #til

**Pull request merged**